src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasisAutoplay.java

```diff
@@ -0,0 +1,83 @@
+package com.wikia.webdriver.testcases.adstests;
+
+import com.wikia.webdriver.common.core.url.Page;
+import com.wikia.webdriver.common.dataprovider.ads.AdsDataProvider;
+import com.wikia.webdriver.common.templates.TemplateNoFirstLoad;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.ad.AutoplayVuap;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+import org.openqa.selenium.Dimension;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+@Test(groups = "AdsVuapAutoplayOasis")
+public class TestAdsVuapOasisAutoplay extends TemplateNoFirstLoad {
+
+  private static final Dimension DESKTOP_SIZE = new Dimension(1920, 1080);
+
+  @Test(groups = "AdsVuapAutoplayAutoplayOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapAutoplayDesktop")
+  public void vuapAutoplayShouldStartPlayingAdvertisementAutomatically(Page page, String slot, String videoIframeSelector) {
+    final AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+    vuap.pause();
+
+    Assert.assertTrue(vuap.hasStarted(), "VUAP did not automatically played when page was opened.");
+    Assert.assertEquals(vuap.findTitle(), "Advertisement", "VUAP video title is not Advertisement.");
+  }
+
+  @Test(groups = "AdsVuapAutoplayStopOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapAutoplayDesktop")
+  public void vuapAutoplayShouldCloseWhenClickingCloseButton(Page page, String slot, String videoIframeSelector) {
+    final AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+    vuap.stop();
+
+    Assert.assertTrue(vuap.isInvisible(), "VUAP should not be visible.");
+  }
+
+  @Test(groups = "AdsVuapAutoplayTimeProgressOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapAutoplayDesktop")
+  public void vuapAutoplayShouldProgressInTime(Page page, String slot, String videoIframeSelector) {
+    final AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "ads" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)